sunpy/lightcurve/lightcurve.py

```diff
@@ -380,6 +381,38 @@ def time_range(self):
         object"""
         return TimeRange(self.data.index[0], self.data.index[-1])
 
+    def concatenate(self, otherlightcurve):
```

Comment:
I don't know if I'm a fan of that API.  The alternative I was trying to describe during the meeting was that a user could use `reduce()` with `concatenate()` as it is currently defined, e.g.:

```python
combined_object = reduce(lambda x, y: x.concatenate(y), input_list)
```